### ISSUE TITLE
[ENH]: possible new cycle type

### ISSUE THREAD

**Issue by TomDonoghue** [neurodsp-tools/neurodsp#227]
This [paper](https://www.frontiersin.org/articles/10.3389/fncom.2016.00087/full) describes a simple type of non-sinuisoidal cycle, which is a bit different from anything we already have available - as it allows for a gaussian like cycle, with a parameter for tweaking the relative amplitude difference of peak / trough. 

The formula, from the paper (formula 9, page 5) is:
`sig = ((cos(2*pi*f*t) + 1) / 2)^b`

Note that everything is pretty standard, except the b parameter, and the plus/divide (which I think just operate to make the data positive, and for scaling). 

This is pretty straight-forward in code:
```
# Set things up
fs = 1000
f = 5
sample = 1500
x = np.arange(sample)

# try new sim
b = 3
vals = ((np.sin(2 * np.pi * f * x / fs) + 1) / 2)**b

# plot sim
plt.plot(x, vals)
```

This gives this outcome:
![Screen Shot 2020-11-19 at 6 44 18 PM](https://user-images.githubusercontent.com/7727566/99737894-415d6480-2a97-11eb-9a3d-23397fa32c82.png)

The parameter `b` is a simple value to change the overall shape of the cycle. 

This is a demonstrated cycle-type in the literature, and is pretty straight-forward, and (I think) extends our functionality, I think we might as well add it. 

@ryanhammonds - when you get a chance, can you:
- have a quick play with this, and compare to our current cycles, especially our existing gaussian cycle, to check / verify that it does add something new and distinct, and then, if so
- prepare an implementation, organizing the formula to match our current organization and API, as we have in `neurodsp/sim/cycles`

**Comment by TomDonoghue** [comment]
I see no reason to not just go for this - but @rdgao might have a more intuitive sense of that formula, and perhaps has quick thoughts and/or wants to give a yay/nay vote?

**Comment by TomDonoghue** [comment]
also, see formula 10, same paper, for a possibly-interesting variant on a sawtooth wave (it's a bit different from what we already have, I think).

**Comment by rdgao** [comment]
these are great! very nice steps towards an analytical description of the different symmetry/shapes.

for eq. 9, yeah the `+1' and `/2` makes the cosine positive and rescaled to [0,1], and the exponent makes the rise to 1 faster. You could even change those to free parameters as well that default to 1 and 0.5, because if you don't do the rescale, the trough get's the same sharpness tuning, and by varying how far the peak and trough are from 1/-1, you can control the sharpness of both. and if we want, we can do a final rescaling to put it into the range of [0,1]. I didn't dig into the literature cited there, but you could do a full-scaled simulation study by varying those parameters to see if they match empirical waveforms.

Eq. 10 is definitely worth implementing as well, since eq.9 will always have symmetrical rise & decay, whereas this one allows for asymmetry.

**Comment by ryanhammonds** [comment]
@TomDonoghue I played with this in a notebook a bit and implemented it in #228. There seem to be pairs of `b` values (`cycle=gaussian_asym`) and `std` values (`cycle=gaussian`) that produce the same plots. There is probably some relationship that to convert between `b` and `std`, but I haven't dug into the math yet.

**Comment by TomDonoghue** [comment]
These have been added in #228  (thanks @ryanhammonds !) so I'm closing this now!
